File: mcp/src/handlers/versions.js
import { readFileSync, existsSync } from 'fs'
import { resolve } from 'path'

export function get_versions(args) {
  const { path: gamePath } = args
  const absPath = resolve(gamePath)
  const configPath = resolve(absPath, '.forkarcade.json')

  if (!existsSync(configPath)) {
    return JSON.stringify({ error: 'No .forkarcade.json found' })
  }

  try {
    const config = JSON.parse(readFileSync(configPath, 'utf-8'))
    return JSON.stringify({
      slug: config.slug,
      title: config.title,
      template: config.template,
      currentVersion: config.currentVersion || 0,
      versions: config.versions || [],
    }, null, 2)
  } catch (e) {
    return JSON.stringify({ error: 'Cannot parse .forkarcade.json' })
  }
}

File: mcp/src/handlers/versions.py
import json
from context import validate_game_path


def get_versions(args):
    game_path = validate_game_path(args["path"])
    config_path = game_path / ".forkarcade.json"

    if not config_path.exists():
        return json.dumps({"error": "No .forkarcade.json found"})

    try:
        config = json.loads(config_path.read_text())
        return json.dumps({
            "slug": config.get("slug"),
            "title": config.get("title"),
            "template": config.get("template"),
            "currentVersion": config.get("currentVersion", 0),
            "versions": config.get("versions", []),
        }, indent=2)
    except Exception:
        return json.dumps({"error": "Cannot parse .forkarcade.json"})
